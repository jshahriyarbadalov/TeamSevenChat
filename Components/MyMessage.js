import styles from '../Styles/Styles';
import React, { Component } from 'react';
import {View, Text, AsyncStorage} from 'react-native';

export class MyMessage extends Component {

    constructor(props) {
        super(props);

        this.state = {
            color: '#ffffff',
        };
    }


    _getItem = async() => {
        try {
            const value = await AsyncStorage.getItem('userColor');
            if (value !== null) {
                this.setState({color: value});
                console.log(value);
            }
        } catch (error) {
            console.log(error);
        }
    };

    componentDidMount() {
        this._getItem();
        const { navigation } = this.props;
        this.focusListener = this.props.navigation.addListener('didFocus', () => {
            this._getItem();
        });
    }

    componentWillUnmount() {
        this.focusListener.remove();
    }

    render() {

        const {color} = this.state;

        return (
            <View style={styles.myMessageRow}>
                <Text style={styles.timeText}>{this.props.time}</Text>
                <View style={styles.myMessageContainer} backgroundColor={color}>
                    <Text style={styles.myMessageText}>{this.props.text}</Text>
                </View>
            </View>
        )
    }
};

export default MyMessage;

