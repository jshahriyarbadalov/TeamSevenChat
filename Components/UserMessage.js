import styles from '../Styles/Styles';
import React, { Component } from 'react';
import { View, Text } from 'react-native';

export class MyMessage extends Component {

    render() {
        return (
            <View style={styles.userMessageRow}>
                <Text style={styles.userTimeText}>{this.props.time}</Text>
                <View style={styles.userMessageContainer} backgroundColor={this.props.background}>
                    <View flex={1} flexDirection="row">
                        <Text style={styles.otherUserName}>{this.props.name}</Text>
                        <Text style={styles.otherUserName}>:</Text>
                        <Text style={styles.otherUserIp}>{this.props.isyour}</Text>
                    </View>
                    <Text style={styles.userMessageText}>{this.props.text}</Text>
                </View>
            </View>
        )
    }
};

export default MyMessage;

