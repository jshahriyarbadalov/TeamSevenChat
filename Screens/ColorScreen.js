import styles from '../Styles/Styles';
import React, { Component } from 'react';
import { View, Image, Text, TouchableOpacity, Dimensions } from 'react-native';
import { AsyncStorage } from 'react-native';


export class ColorScreen extends Component {

    constructor(props) {
        super(props);

        const isPortrait = () => {
            const dim = Dimensions.get('screen');
            return dim.height >= dim.width;
        };

        this.state = {
            orientation: isPortrait() ? 'portrait' : 'landscape'
        };

        // Event Listener for orientation changes
        Dimensions.addEventListener('change', () => {
            this.setState({
                orientation: isPortrait() ? 'portrait' : 'landscape'
            });
        });
    }


    _storeColor = async (color) => {
        try {
            await AsyncStorage.setItem('userColor', color);
        } catch (error) {
            console.log(error)
        }
    };

    render() {

        if (this.state.orientation === 'portrait') {
            return (
                <View style={{alignItems: 'center', backgroundColor: '#fafafa', flex: 1}}>
                    <View style={{justifyContent: 'center', alignItems: 'center'}}>
                        <Text style={styles.colorsText}>Выбор цвета для текстового поля:</Text>
                    </View>
                    <View style={styles.colorsContainer}>
                        <View style={styles.colorInner}>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#e15f5f')
                            }}>
                                <Image source={require('../images/circle.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#5296a5')
                            }}>
                                <Image source={require('../images/circle-2.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#89db7d')
                            }}>
                                <Image source={require('../images/circle-3.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                        </View>
                        <View style={styles.colorInner}>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#e7e568')
                            }}>
                                <Image source={require('../images/circle-5.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#ffffff')
                            }}>
                                <Image source={require('../images/circle-6.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#c19d48')
                            }}>
                                <Image source={require('../images/circle-7.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                        </View>
                        <View style={styles.colorInner}>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#33cccc')
                            }}>
                                <Image source={require('../images/circle-9.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#3955eb')
                            }}>
                                <Image source={require('../images/circle-8.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#857ddb')
                            }}>
                                <Image source={require('../images/circle-4.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                        </View>
                        <View style={styles.colorInner}>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#ff9966')
                            }}>
                                <Image source={require('../images/circle-10.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#ffff66')
                            }}>
                                <Image source={require('../images/circle-11.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                            <TouchableOpacity onPress={() => {
                                this._storeColor('#aeaeae')
                            }}>
                                <Image source={require('../images/circle-12.png')} style={styles.colorCircle}/>
                            </TouchableOpacity>
                        </View>
                    </View>
                </View>
            )
        } else {
            return (
                <View style={{alignItems: 'center', backgroundColor: '#fafafa', flex: 1, flexDirection: 'row'}}>
                    <View flex={2} flexDirection="column">
                        <View style={{justifyContent: 'center', alignItems: 'center', marginTop: 0}}>
                            <Text style={styles.colorsText}>Выбор цвета для текстового поля:</Text>
                        </View>
                    </View>
                    <View flex={2} flexDirection="column">
                        <View style={styles.colorsContainer}>
                            <View style={styles.colorInner}>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#e15f5f')
                                }}>
                                    <Image source={require('../images/circle.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#5296a5')
                                }}>
                                    <Image source={require('../images/circle-2.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#89db7d')
                                }}>
                                    <Image source={require('../images/circle-3.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                            </View>
                            <View style={styles.colorInner}>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#e7e568')
                                }}>
                                    <Image source={require('../images/circle-5.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#ffffff')
                                }}>
                                    <Image source={require('../images/circle-6.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#c19d48')
                                }}>
                                    <Image source={require('../images/circle-7.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                            </View>
                            <View style={styles.colorInner}>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#33cccc')
                                }}>
                                    <Image source={require('../images/circle-9.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#3955eb')
                                }}>
                                    <Image source={require('../images/circle-8.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#857ddb')
                                }}>
                                    <Image source={require('../images/circle-4.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                            </View>
                            <View style={styles.colorInner}>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#ff9966')
                                }}>
                                    <Image source={require('../images/circle-10.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#ffff66')
                                }}>
                                    <Image source={require('../images/circle-11.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                                <TouchableOpacity onPress={() => {
                                    this._storeColor('#aeaeae')
                                }}>
                                    <Image source={require('../images/circle-12.png')} style={styles.colorCircle}/>
                                </TouchableOpacity>
                            </View>
                        </View>
                    </View>
                </View>
            )
        }
    }
};

export default ColorScreen;

