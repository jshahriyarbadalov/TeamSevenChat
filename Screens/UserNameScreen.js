import styles from '../Styles/Styles';
import React, { Component } from 'react';
import { SafeAreaView, ScrollView, View, Image, Text, TouchableOpacity, TextInput, AsyncStorage, KeyboardAvoidingView, Platform, Dimensions } from 'react-native';


const URL = 'http://web-chat.eu-4.evennode.com/putchangeuser';
const urlPut = 'http://web-chat.eu-4.evennode.com/putuser';
const urlAllUsers = 'http://web-chat.eu-4.evennode.com/getusers';

export class UserNameScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            correct: true,
            error: '',
            name: '',
            inputText: '',
            userNames: [],
            data: [],
        };

        const isPortrait = () => {
            const dim = Dimensions.get('screen');
            return dim.height >= dim.width;
        };

        this.state = {
            orientation: isPortrait() ? 'portrait' : 'landscape'
        };

        // Event Listener for orientation changes
        Dimensions.addEventListener('change', () => {
            this.setState({
                orientation: isPortrait() ? 'portrait' : 'landscape'
            });
        });
    }

    ws = new WebSocket(URL);

    _storeUser = async (name) => {
        try {
            await AsyncStorage.setItem('userName', name);
        } catch (error) {
            console.log(name)
        }
    };

    _getName = async () => {
        try {
            const value = await AsyncStorage.getItem('userName');
            if (value !== null) {
                this.setState({ name: value });
            }
        } catch (error) {
            console.log(error);
        }
    };

    getUsers() {
        return fetch(urlAllUsers)
            .then((response) => response.json())
            .then((responseJson) => {
                this.setState({ data: responseJson });
                this.userNames();
            })
            .catch((error) => {
                console.error(error);
            });

    }

    userNames = () => {
        let data = this.state.data;
        let userNames = [];

        data.forEach(element => {
            userNames.push(element.name)
        });

        this.setState({ userNames: userNames })
    };

    componentDidMount() {
        this._getName();
        this.getUsers();
        this.setState({correct: true, inputText: '', error: false})
    }

    addName = (inputText) => {
        this.setState({ inputText: inputText, correct: true });
    };

    submitName() {
        let foundNamePair = this.state.userNames.find(obj => { return obj === this.state.inputText });

        if (this.state.inputText !== "" && this.state.inputText.trim().length !== 0) {
            if (this.state.inputText.length >= 3 && this.state.inputText.length <= 10) {
                if (foundNamePair !== undefined) {
                    this.setState({ error: "*это имя уже занято, введите другое", correct: false })
                } else {
                    fetch(URL, {
                        method: 'PUT',
                        headers: {
                            'Accept': 'application/json',
                            'Content-Type': 'application/json;charset=UTF-8'
                        },
                        body: JSON.stringify({
                            name: this.state.inputText,
                            state: true,
                        }),
                    });

                    this.setState({ inputText: "", correct: true });
                    this.props.navigation.navigate('ChatScreen');
                    this.setState({ error: "" });
                    this._storeUser(this.state.inputText);
                }

            } else {
                this.setState({ error: "*должно быть мин 3 символа макс 10 ", correct: false })

            }

        } else {
            this.setState({ error: "*введите имя пользователя", correct: false })
        }

    }


    press() {
        return this.submitName();
    }

    cancelBtn() {
        this.setState({ inputText: "", correct: true });
        this.props.navigation.navigate('ChatScreen');
        this.setState({ error: "", correct: true })

    }

    render() {
        const isCorrect = this.state.correct;

        if (this.state.orientation === 'portrait') {
            return (
                <SafeAreaView>
                    <View style={{alignItems: 'center'}}>
                        <View>
                            <KeyboardAvoidingView
                                style={{flex: 1,}} behavior='padding'
                                keyboardVerticalOffset={
                                    Platform.select({
                                        ios: () => 110,
                                        android: () => -150
                                    })()
                                }>
                                <View style={{justifyContent: 'center', alignItems: 'center'}}>
                                    <Text style={styles.nameChangeText}>Введите имя пользователя:</Text>

                                    <TextInput placeholder={this.state.name} style={styles.changeNameTextInput}
                                               borderColor={isCorrect ? '#666' : 'red'}
                                               onChangeText={(inputText) => this.addName(inputText.trim())}
                                               clearButtonMode="always" autoCorrect={false}/>
                                    <Text style={styles.validNameText}>{this.state.error}</Text>
                                </View>
                                <View style={styles.nameChangeButtons}>
                                    <TouchableOpacity style={styles.changedNameCancelButton}
                                                      onPress={this.cancelBtn.bind(this)}>
                                        <Text style={styles.registrationBtnText}>Отмена</Text>
                                    </TouchableOpacity>
                                    <TouchableOpacity style={styles.changedNameSaveButton}
                                                      onPress={() => this.press()}>
                                        <Text style={styles.loginBtnText}>ОК</Text>
                                    </TouchableOpacity>
                                </View>
                            </KeyboardAvoidingView>
                        </View>
                    </View>
                </SafeAreaView>
            )
        } else {
            return (
                <SafeAreaView>
                    <View style={{alignItems: 'center'}}>
                        <View>
                            <KeyboardAvoidingView
                                style={{flex: 1,}} behavior='padding'
                                keyboardVerticalOffset={
                                    Platform.select({
                                        ios: () => 110,
                                        android: () => -150
                                    })()
                                }>
                                <View style={{justifyContent: 'center', alignItems: 'center'}}>
                                    <Text style={styles.nameChangeTextLand}>Введите имя пользователя:</Text>

                                    <TextInput placeholder={this.state.name} style={styles.changeNameTextInputLand}
                                               borderColor={isCorrect ? '#666' : 'red'}
                                               onChangeText={(inputText) => this.addName(inputText.trim())}
                                               clearButtonMode="always" autoCorrect={false}/>
                                    <Text style={styles.validNameText}>{this.state.error}</Text>
                                </View>
                                <View style={styles.nameChangeButtons}>
                                    <TouchableOpacity style={styles.changedNameCancelButtonLand}
                                                      onPress={this.cancelBtn.bind(this)}>
                                        <Text style={styles.registrationBtnText}>Отмена</Text>
                                    </TouchableOpacity>
                                    <TouchableOpacity style={styles.changedNameSaveButtonLand}
                                                      onPress={() => this.press()}>
                                        <Text style={styles.loginBtnText}>ОК</Text>
                                    </TouchableOpacity>
                                </View>
                            </KeyboardAvoidingView>
                        </View>
                    </View>
                </SafeAreaView>
            )
        }
    }
};



export default UserNameScreen;

