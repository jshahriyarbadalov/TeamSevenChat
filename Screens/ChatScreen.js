import styles from '../Styles/Styles';
import React, { Component } from 'react';
import { SafeAreaView, ScrollView, View, Image, Text, TouchableOpacity, TextInput, FlatList, Platform, Alert, KeyboardAvoidingView, AsyncStorage } from 'react-native';
import MyMessage from '../Components/MyMessage';
import UserMessage from '../Components/UserMessage';
import ReconnectingWebSocket from 'react-native-reconnecting-websocket';
import PushNotification from "react-native-push-notification";



const URL = 'ws://web-chat.eu-4.evennode.com/';
const urlPut = 'http://web-chat.eu-4.evennode.com/putuser';

PushNotification.configure({

  onRegister: function (token) {
    console.log("TOKEN:", token);
  },

  onNotification: function (notification) {
    console.log("NOTIFICATION:", notification);
    notification.finish(PushNotificationIOS.FetchResult.NoData);
  },

  senderID: "246105469934",

  permissions: {
    alert: true,
    badge: true,
    sound: true
  },

  popInitialNotification: true,
  requestPermissions: true
});


export default class ChatScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      ModalV: false,
      isLoading: true,
      messages: [],
      inputText: '',
      time: new Date().toLocaleTimeString(navigator.language, { hour: '2-digit', minute: '2-digit' }),
      ws: '',
      IP: '',
      name: '',
      isyour: '',
      myname: '',
    };

    this.messageColors = [];
  }

  _getName = async () => {
    try {
      const value = await AsyncStorage.getItem('userName');
      if (value !== null) {
        this.setState({ myname: value });
      }
    } catch (error) {
      console.log(error);
    }
  };

  getIp() {
    fetch('http://web-chat.eu-4.evennode.com/getIp', {
      method: 'GET',
    })
        .then(response => response.json())
        .then(responseJson => {
          this.setState({ IP: responseJson.ip });
        })
        .catch(error => {
          console.error(error);
        });
  }

  getNotification(name, text) {

    PushNotification.localNotification({
      title: name,
      message: text,
    });

    PushNotification.configure({
      onNotification: function (notification) {
        console.log('Notification is clicked')
      }
    })
  }

  ws = new ReconnectingWebSocket(URL)
  componentDidMount() {

    this._getName();

    fetch('http://web-chat.eu-4.evennode.com/getmessages', {
      method: 'GET',
    })
        .then(response => response.json())
        .then(responseJson => {
          this.setState({ isLoading: false, messages: responseJson.reverse(), name: responseJson.name, isyour: responseJson.ip });
        })
        .catch(error => {
          console.error(error);
        });

    this.getIp();

    this.ws.onopen = () => {
      console.log('connected')
    };

    this.ws.onmessage = evt => {
      const message = JSON.parse(evt.data)
      this.addMessage(message)
      if (this.state.IP !== message.ip) {
        this.getNotification(message.name, message.text);
      }
    };

    this.ws.onclose = () => {
      console.log('disconnected')
      this.setState({
        ws: new WebSocket(URL),
      });

    }
  }


  addMessage = message =>
      this.setState(state => ({ messages: [message, ...this.state.messages] }))

  submitMessage() {
    if (this.state.inputText !== "" && this.state.inputText.trim().length !== 0) {
      this.ws.send(JSON.stringify({
        text: this.state.inputText.trim(),
        time: this.state.time,
        ip: this.state.IP,
        name: this.state.myname,

      }));

      this.setState({ inputText: '' })

      fetch("http://web-chat.eu-4.evennode.com/putmessage", {
        method: "put",
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json;charset=UTF-8'
        },

        body: JSON.stringify({
          text: this.state.inputText.trim(),
          time: this.state.time,
          ip: this.state.IP,
          name: this.state.myname,
        })
      })
          .then((response) => {
            console.log(response)
          })

    } else {
      this.setState({ inputText: '' })

    }

  }

  getRandomColor = () => {

    let colors = ['#BF80FF', '#cc99ff', '#ffcccc', '#ff80d5', '#ffcccc', '#80ffbf', '#b3ccff', '#b3b3ff', '#ffcce6', '#e6b800', '#b3cccc', '#ff9980', '#99b3ff', '#6699ff', '#99bbff'];
    let color = '';
    for (let i = 0; i <= 15; i++) {
      color = colors[Math.floor(Math.random() * 16)]
    }
    return color;
  };

  render() {
    return (
        <SafeAreaView style={styles.chatScreenContainer}>
          <KeyboardAvoidingView style={{ flex: 1, }} behavior='padding'
                                keyboardVerticalOffset={
                                  Platform.select({
                                    ios: () => 110,
                                    android: () => -150
                                  })()
                                }>
            <View style={styles.chatScreenContainer}>
              <View style={styles.messagesMainContainer}>
                <View style={styles.dinamicMessagesField}>
                  <View style={styles.innerMessagesField}>
                    <FlatList
                        inverted={true}
                        data={this.state.messages}
                        renderItem={({ item }) => {
                          let color = '#666666'; //default color
                          let foundColorPair = this.messageColors.find(obj => { return obj.ip === item.ip });
                          if (foundColorPair !== undefined) {
                            color = foundColorPair.color;
                          } else {
                            color = this.getRandomColor();
                            this.messageColors.push({ ip: item.ip, color: this.getRandomColor() });
                          }

                          if (item.time !== undefined) {
                            if (item.name !== this.state.myname) {
                              return (<UserMessage text={item.text} background={color} key={item.text} name={item.name}
                                                   isyour={item.ip}
                                                   time={new Date(item.time).toLocaleTimeString(navigator.language, {
                                                     hour: '2-digit',
                                                     minute: '2-digit'
                                                   })} />)
                            } else {
                              return (<MyMessage navigation={this.props.navigation} text={item.text} background={color}
                                                 key={item.text}
                                                 time={new Date(item.time).toLocaleTimeString(navigator.language, {
                                                   hour: '2-digit',
                                                   minute: '2-digit'
                                                 })} />)
                            }
                          }
                        }}
                        keyExtractor={(item) => item.text + item.time + item.ip}

                    />

                  </View>
                </View>
              </View>
              <View style={styles.messagesInputContainer}>
                <TextInput
                    placeholder="Введите сообщение..."
                    style={styles.messagesInput}
                    multiline={true}
                    numberOfLines={4}
                    placeholderTextColor="#000000"
                    onChangeText={inputText => this.setState({ inputText })}
                    value={this.state.inputText}
                    maxLength={400}
                />
                <TouchableOpacity
                    style={styles.paperPlaneContainer}
                    onPress={this.submitMessage.bind(this)}>
                  <Image
                      source={require('../images/paper-plane.png')}
                      style={styles.paperPlaneButton}
                  />
                </TouchableOpacity>
              </View>
            </View>
          </KeyboardAvoidingView>
        </SafeAreaView>
    );
  }
}
