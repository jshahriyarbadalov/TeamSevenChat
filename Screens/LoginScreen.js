import styles from '../Styles/Styles';
import React, { Component } from 'react';
import { SafeAreaView, View, Image, Text, TouchableOpacity, TextInput, KeyboardAvoidingView, AsyncStorage, Dimensions } from 'react-native';
import ChatScreen from './ChatScreen';

const url = 'http://web-chat.eu-4.evennode.com/getip';
const urlPut = 'http://web-chat.eu-4.evennode.com/putuser';


export class LoginScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            ip: '',
            name: '',
            users: [],
            myname: '',
        };

        const isPortrait = () => {
            const dim = Dimensions.get('screen');
            return dim.height >= dim.width;
        };

        this.state = {
            orientation: isPortrait() ? 'portrait' : 'landscape'
        };

        // Event Listener for orientation changes
        Dimensions.addEventListener('change', () => {
            this.setState({
                orientation: isPortrait() ? 'portrait' : 'landscape'
            });
        });
    }

    getUserIp() {
        return fetch(url)
            .then((response) => response.json())
            .then((responseJson) => {
                this.setState({ ip: responseJson.ip });
            })
            .catch((error) => {
                console.error(error);
            });
    }

    getUserName() {
        fetch(urlPut, {
            method: 'PUT',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json;charset=UTF-8'
            },
            body: JSON.stringify({
                id: this.state.id,
                state: true,
            }),
        }).then((response) => response.json())
            .then((responseJson) => {
                this.setState({ myname: responseJson })
            })
    }

    _getName = async () => {
        try {
            const value = await AsyncStorage.getItem('userName');
            if (value !== null) {
                this.setState({ name: value });
                console.log(value);
            }
        } catch (error) {
            console.log(error);
        }
    };

    componentDidMount() {
        this.getUserIp();
        this._getName();
        this.getUserName();
    }

    render() {
        const { ip, myname, name } = this.state;
        if (this.state.orientation === 'portrait') {
            console.log(myname)
            if (myname !== 0) {
                return (
                    <SafeAreaView style={styles.welcomeScreenContainer}>
                        <KeyboardAvoidingView
                            style={{ flex: 1, }} behavior='padding'
                            keyboardVerticalOffset={
                                Platform.select({
                                    ios: () => 110,
                                    android: () => -150
                                })()
                            }>
                            <View style={styles.welcomeScreenInner}>
                                <Image source={require('../images/logo.png')} style={styles.loginScreenLogo} />
                                <Text style={styles.loginIpTitle}>
                                    Ваш IP:
                                </Text>
                                <Text style={styles.loginIpNumber}>
                                    {ip}
                                </Text>
                                <Text style={styles.loginApproveText}>
                                    Вы зарегестрированный пользователь, введите пожалуйста свое имя:
                                </Text>
                                <TextInput value={ name } style={styles.loginTextInput} />
                                <TouchableOpacity style={styles.loginButton}
                                    onPress={() => this.props.navigation.navigate('ChatScreen')}>
                                    <Text style={styles.loginBtnText}>Войти</Text>
                                </TouchableOpacity>
                            </View>
                        </KeyboardAvoidingView>
                    </SafeAreaView>
                )
            } else {
                return (
                    <SafeAreaView style={styles.welcomeScreenContainer}>
                        <View style={styles.welcomeScreenInner}>
                            <Image source={require('../images/logo.png')} style={styles.loginScreenLogo} />
                            <Text style={styles.loginIpTitle}>
                                Ваш IP:
                            </Text>
                            <Text style={styles.loginIpNumber}>
                                {ip}
                            </Text>
                            <Text style={styles.loginApproveText}>
                                Вы не зарегестрированный пользователь, пожалуйста, зарегестрируйтесь
                            </Text>
                            <TouchableOpacity style={styles.registrationButton}
                                onPress={() => this.props.navigation.navigate('RegistrationScreen')}>
                                <Text style={styles.registrationBtnText}>Регистрация</Text>
                            </TouchableOpacity>
                        </View>
                    </SafeAreaView>
                )
            }
        } else {
            if (myname !== 0) {
                return (
                    <SafeAreaView style={styles.welcomeScreenContainer}>
                        <KeyboardAvoidingView
                            style={{ flex: 1, }} behavior='padding'
                            keyboardVerticalOffset={
                                Platform.select({
                                    ios: () => 110,
                                    android: () => -150
                                })()
                            }>
                            <View style={styles.welcomeScreenInner} flex={1} flexDirection='row'>
                                <View flex={2} flexDirection='column'>
                                    <View style={{justifyContent: 'center', alignItems: 'center'}}>
                                        <Image source={require('../images/logo.png')} style={styles.loginScreenLogoLand} />
                                        <Text style={styles.loginIpTitleLand}>
                                            Ваш IP:
                                        </Text>
                                        <Text style={styles.loginIpNumber}>
                                            {ip}
                                        </Text>
                                        <Text style={styles.loginApproveTextLand} >
                                            Вы зарегестрированный пользователь, введите пожалуйста свое имя:
                                        </Text>
                                    </View>
                                </View>
                                <View flex={2} flexDirection='column'>
                                    <TextInput value={ name } style={styles.loginTextInput} />
                                    <TouchableOpacity style={styles.loginButton}
                                        onPress={() => this.props.navigation.navigate('ChatScreen')}>
                                        <Text style={styles.loginBtnText}>Войти</Text>
                                    </TouchableOpacity>
                                </View>
                            </View>
                        </KeyboardAvoidingView>
                    </SafeAreaView>
                )
            } else {
                return (
                    <SafeAreaView style={styles.welcomeScreenContainer}>
                        <View style={styles.welcomeScreenInner}>
                            <Image source={require('../images/logo.png')} style={styles.loginScreenLogo} />
                            <Text style={styles.loginIpTitle}>
                                Ваш IP:
                            </Text>
                            <Text style={styles.loginIpNumber}>
                                {ip}
                            </Text>
                            <Text style={styles.loginApproveText} >
                                Вы не зарегестрированный пользователь, пожалуйста, зарегестрируйтесь
                            </Text>
                            <TouchableOpacity style={styles.registrationButton}
                                onPress={() => this.props.navigation.navigate('RegistrationScreen')}>
                                <Text style={styles.registrationBtnText}>Регистрация</Text>
                            </TouchableOpacity>
                        </View>
                    </SafeAreaView>
                )
            }
        }
    }
};

export default LoginScreen;

