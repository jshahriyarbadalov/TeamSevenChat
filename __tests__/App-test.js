/**
 * @format
 */

import 'react-native';
import React from 'react';
import MyMessage from '../Components/MyMessage';
import UserMessage from '../Components/UserMessage';
import ChatScreen from '../Screens/ChatScreen';
import ColorScreen from '../Screens/ColorScreen';
import LoginScreen from '../Screens/LoginScreen';
import RegistrationScreen from '../Screens/RegistrationScreen';
import UserNameScreen from '../Screens/UserNameScreen';
import UsersListScreen from '../Screens/UsersListScreen';
import WelcomeScreen from '../Screens/WelcomeScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';




describe('Test my Ui Components', ()=>{

    it('snapshot testing MyMessage component ', ()=>{
      const myCompMessage=renderer.create(<MyMessage/>).toJSON
      expect(myCompMessage).toMatchSnapshot();
    });

    it('snapshot testing UserMessage component ', ()=>{
      const myCompUM=renderer.create(<UserMessage/>).toJSON
      expect(myCompUM).toMatchSnapshot();
    });

    it('snapshot testing ChatScreen component ', ()=>{
      const myCompChat=renderer.create(<ChatScreen/>).toJSON
      expect(myCompChat).toMatchSnapshot();
    });

    it('snapshot testing ColorScreen component ', ()=>{
      const myCompColor=renderer.create(<ColorScreen/>).toJSON
      expect(myCompColor).toMatchSnapshot();
    });

    it('snapshot testing LoginScreen component', ()=>{
      const myCompLogin=renderer.create(<LoginScreen/>).toJSON
      expect(myCompLogin).toMatchSnapshot();
    });
  
    it('snapshot testing RegistrationScreen component', ()=>{
      const myCompRegistr=renderer.create(<RegistrationScreen/>).toJSON
      expect(myCompRegistr).toMatchSnapshot();
    });

    it('snapshot testing UserNameScreen component', ()=>{
      const myCompUserName=renderer.create(<UserNameScreen/>).toJSON
      expect(myCompUserName).toMatchSnapshot();
    });

    it('snapshot testing UserListScreen component', ()=>{
      const myCompUsersList=renderer.create(<UsersListScreen/>).toJSON
      expect(myCompUsersList).toMatchSnapshot();
    });

    it('snapshot testing WelcomeScreen component', ()=>{
      const myCompWelcome =renderer.create(<WelcomeScreen/>).toJSON
      expect(myCompWelcome).toMatchSnapshot();
    });
});
