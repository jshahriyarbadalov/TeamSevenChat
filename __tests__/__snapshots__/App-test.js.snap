// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test my Ui Components snapshot Testing MyMessage component  1`] = `[Function]`;

exports[`Test my Ui Components snapshot Testing MyMessage component  2`] = `[Function]`;

exports[`Test my Ui Components snapshot testing ChatScreen component  1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing ColorScreen component  1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing LoginScreen component 1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing MyMessage component  1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing RegistrationScreen component 1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing UserListScreen component 1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing UserMessage component  1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing UserNameScreen component 1`] = `[Function]`;

exports[`Test my Ui Components snapshot testing WelcomeScreen component 1`] = `[Function]`;
